tests/unit/api1_test.py

```diff
@@ -180,10 +185,21 @@ def test_custom_login_flow(
         ),
     )
 
+    backend_3 = QuantinuumBackend(
+        device_name=fake_device,
```

Comment:
Do we want to test the fake device here? Not sure I fully understood this testcase?

Reply:
The `fake_device` is just a dummy string from the `mock_machine_info` fixture, it doesn't matter because we catch all outward calls with `requests_mock`.

The testcase here is to make sure that the `login` endpoint is called three times (by backend_{1,2,3}) with `assert login_route.call_count == 3`.  `backend_4` shouldn't call `login` because `backend_3` has cached the authentication tokens in storage.

Hope this makes sense

Reply:
Yes, thank you! Sounds good